### PR TITLE
Use isemail for email validation in user model

### PR DESCRIPTION
**Pull request by vdust**
instead of the regular expression that did not correctly validate all
email addresses (both false positives and false negatives).

Should fix issue #1723

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by superkhau** [comment]
@rmg There seems to be an issue with io.js. Can you take a look?

**Comment by rmg** [comment]
@slnode ok to test

**Comment by rmg** [comment]
The `.travis.yml` file needs to be updated to replace `iojs` with `4`.

**Comment by bajtos** [comment]
+1 for using `isemail` instead of a hand-written regexp.

File: test/user.test.js [hunk 1]
```diff
@@ -91,7 +91,7 @@ describe('User', function() {
         assert.equal(err.details.context, User.modelName);
         assert.deepEqual(err.details.codes.email, [
           'presence',
-          'format.null'
+          'custom'
```
Comment (bajtos) [inline test/user.test.js]: Could you please improve your patch to report a unique error code, e.g. `isemail`? The validation codes are typically used by front-end to display a localised error message. The code `custom` can mean anything...

Comment (vdust) [inline test/user.test.js]: I'm not sure how to do this. The Validatable api in loopback-datasource-juggler/lib/validations.js doesn't seem to allow override of the validator name. Should i move the logic in a new Validatable.validateEmail function there and then simply use it in the user model ?

**Comment by bajtos** [comment]
Closing in favour of #2716.

**Pull request closed without merging**